README.md

```diff
@@ -94,6 +97,183 @@ see [Developer’s guide to auth with Azure Resource Manager API](http://aka.ms/
+
+## What is a  Profile?
+
+A profile is a combination of different resource types with different versions from different services. Using a profile,
+will help you mix and match between various resource types.
+
+## What to use?
+* If you would like to use the latest versions of **all** the services, then the recommendation is to use the **latest** profile of the Azure SDK rollup gem.
+
+* If you would like to use the services compatible with the **azure stack**, then the recommendation is to use the **v2017_03_09** profile of the Azure SDK rollup gem.
+
+* If you would like to use the latest version of **some** services, then the recommendation is to use the **latest** profile of those specific gems.
+
+* If you would like to use **specific** versions, then the recommendation is to use the **multiple API** versions defined inside each gem.
+
+**Note** All the above options could be combined within the same application.
+
+## Usage of Azure SDK gem
+
+Azure SDK gem is a rollup of all the supported gems in the Ruby SDK. This gem consists of a **latest** profile which supports the latest version of all services. In addition, this also has a **v2017_03_09** profile which is built for azure stack.
+
+### Install
+
+You can install the rollup gem with the following command:
+
+```ruby
+gem install 'azure_sdk'
+```
+
+### Existing Profiles
+
+Currently, the azure_sdk rollup gem has two profiles.
+  1. v2017_03_09 (Profile built for azure stack)
+  2. latest (Profile consists of Latest versions of all services)
+
+You could choose the profile that you would like to use. If you would like to use the latest versions of **all** the services, then the recommendation is to use the **latest** profile of the Azure SDK rollup gem.
+
+If you would like to use the services compatible with the **azure stack**, then the recommendation is to use the **v2017_03_09** profile of the Azure SDK rollup gem. 
+
+### Usage
+
+The following lines should be used to instantiate a profile client:
+
+```ruby
+provider = MsRestAzure::ApplicationTokenProvider.new(
+       ENV['AZURE_TENANT_ID'],
+       ENV['AZURE_CLIENT_ID'],
+       ENV['AZURE_CLIENT_SECRET'])
+credentials = MsRest::TokenCredentials.new(provider)
+
+options = {
+  credentials: credentials,
+  subscription_id: ENV['AZURE_SUBSCRIPTION_ID']
+}
+
+profile_client = Azure::Profiles::V2017_03_09::Mgmt::Client.new(options)
+```
+
+The profile client could be used to access individual RPs:
+
+ ```ruby
+# To access the operations associated with Compute
+profile_client.compute.virtual_machines.get 'RESOURCE_GROUP_NAME', 'VIRTUAL_MACHINE_NAME'
+
+# Option 1: To access the models associated with Compute
+purchase_plan_obj = profile_client.compute.model_classes.purchase_plan.new
+
+# Option 2: To access the models associated with Compute
+purchase_plan_obj = Azure::Profiles::V2017_03_09::Compute::Mgmt::Models::PurchasePlan.new
+
+```
```

Comment:
maybe we should add a note at the end like: "Notice namespace `Azure::Profiles::<Profile_Name>::<Service_Name>` "

Reply:
Done